src/bluesky/callbacks/olog.py

```diff
@@ -173,5 +173,5 @@ def start(self, doc):
             pformat(doc),
         )
         olog_status = self.client.log(document_content, logbooks=self.logbook)
-        logger.debug("client.log returned %s" % olog_status)
+        logger.debug(f"client.log returned {olog_status}")
```

Comment:
```suggestion
        logger.debug("client.log returned %s", olog_status)
```

Reply:
I've only made this change because this line (and 5 others) didn't pass the linter checks. I'm happy to just silence it with `# noqa: UP031`, but I am curious what is the reason we can't use f-strings here since, as far as I understand, we declare support for only Python 3.8+? Are there any back-compatibility requirements that I'm missing?

Reply:
See, for example, this [article](https://dev.to/izabelakowal/what-is-the-best-string-formatting-technique-for-logging-in-python-d1d).

The methods of the logging package have different arguments than `print()`.  The first argument is the format string.  Subsequent arguments are various python objects to be used with the format string _if_ the method is to generate its output.  By using an f-string (which is allowed since it suffices as a format string), you are circumventing the decision process in the logging method.  That process can choose not to create an output based on the logging level.

tl;dr:  f-strings in logging defeat an optimization based on the log level

Reply:
BTW, `"%s" % item` is the same as `f"{item}"`, should use `"%s", item` instead.  It's the comma that cooperates with the optimization in logging.

Reply:
ah right, thanks! I haven't realized it was in `logging.debug`, not in `print`. totally makes sense!

Reply:
I expect the linter would flag `"%s" % item` for improvement.